core/computed/resource-summary.js

```diff
@@ -32,14 +32,42 @@ class ResourceSummary {
     return requestToResourceType[record.resourceType] || 'other';
   }
 
+  /**
+   * Although the browser will only need one favicon asset, we don't know which
+   * one will actually be requested. The browser could request multiple if initial
+   * candidates are found to be unusable. This function simply returns the set of all
+   * possible favicon urls.
+   * @see https://html.spec.whatwg.org/multipage/links.html#linkTypes
+   * @param {LH.Artifacts.LinkElement[]} LinkElements
+   * @param {string|undefined} mainDocumentUrl
+   * @return {string[]}
+   */
+  static determineFaviconUrls(LinkElements, mainDocumentUrl) {
+    if (!mainDocumentUrl) return [];
```

Comment:
Now that we don't need `mainDocumentUrl` for L55, we shouldn't exit early if there are icon link elements for us to go through.